pyfstat/simulate.py

```diff
@@ -0,0 +1,101 @@
+import logging
+from typing import Optional, Union
+
+import lal
+import lalpulsar
+import numpy as np
+from attrs import converters, define, field
+
+from pyfstat.snr import DetectorStates, SignalToNoiseRatio
+from pyfstat.utils import get_ephemeris_files
+
+logger = logging.getLogger(__name__)
+
+
+@define(kw_only=True, slots=False)
+class MakeFakeData:
+
+    ephemeris: lalpulsar.EphemerisData = field(
+        factory=lambda: lalpulsar.InitBarycenter(*get_ephemeris_files())
+    )
+
+    def set_data_params(
+        self,
+        fMin: float,
+        Band: float,
+        Tsft: int,
+        timestamps: dict,
+        sqrtSX: dict,
+        randSeed: int = 0,
+    ):
+
+        self.data_params = lalpulsar.CWMFDataParams()
+        self.data_params.randSeed = randSeed
+        self.data_params.fMin = fMin
+        self.data_params.Band = Band
+
+        multi_timestamps, multi_detector = DetectorStates._parse_timestamps_and_detectors(
+            timestamps,
+            Tsft,
+            detectors=None,
+        )
+
+        # FIXME This doesn't fail here, why does it fail at test time????
+        logger.error(f"{multi_timestamps.data[0].data[0]}")
+        self.data_params.multiTimestamps = multi_timestamps
+        logger.error(f"{self.data_params.multiTimestamps.data[0].data[0]}")
+        self.data_params.multiIFO = multi_detector
```

Comment:
This works.

Reply:
Well, this works because `self.data_params.multiTimestamps` points to `multi_timestamps` by reference.
Once this scope is left, `multi_timestamps` doesn't exist anymore, some reference to deeper levels of the structure exists and seg fault raises.

Reply:
I don't see any 1D_vector swig declarations in [the source code](https://lscsoft.docs.ligo.org/lalsuite/lalpulsar/_c_w_make_fake_data_8h_source.html#l00069), so that could probably be it...?